### ISSUE TITLE
XmlElement -> xml string

### ISSUE THREAD

**Issue by Freymaurer** [Zaid-Ajaj/Fable.SimpleXml#21]
It would be nice to have a reverse function for `SimpleXml.parseElement`, as my current workflow is:

- Have xml string (`string`), parse it with `SimpleXml.parseElement` (`XmlElement`).
- Update some of the children (`XmlElement`).
- Next i want to use the updated `XmlElement` to create a xml string (`string`) to also update the origin.

I wrote the following function for my use-case, but my experience with xml is rather limited and my solution does not consider:
- XmlElement.Namespace 
- XmlElement.IsComment
- XmlElement.isTextNode
- Multiple Root elements

```fsharp
open Fable.SimpleXml
open Fable.SimpleXml.Generator

let xmlElementToXmlString (root:XmlElement) =
    let rec createChildren (child:XmlElement) =
        match child.SelfClosing with
        | true ->
            leaf child.Name [
                for cAttr in child.Attributes do
                    yield attr.value(cAttr.Key,cAttr.Value)
            ]
        | false ->
            node child.Name [
                for cAttr in child.Attributes do
                    yield attr.value(cAttr.Key,cAttr.Value)
            ][
                for grandChild in child.Children do
                    yield createChildren grandChild
                yield
                    text child.Content
            ]
    node root.Name [
        for rAttr in root.Attributes do
            yield attr.value(rAttr.Key,rAttr.Value)
    ] [
        for child in root.Children do
            yield createChildren child
        yield
            text root.Content
    ] |> serializeXml
```

**Comment by Zaid-Ajaj** [comment]
Sounds like a nice addition @Freymaurer thanks for the example. I leave this up for grabs if anyone is interested in the functionality

**Comment by Freymaurer** [comment]
Hey i want to help out and add it, but i don't know how to set up a testing environment. As this would be the first time i contribute to someone elses project and i am not very experienced i might need rather in-depth instructions for this.

For now i added:

- XNode.Comment
- XNode.Namespace
- Reverse Parsing function for one and many root elements.

Still missing:

- [ ] Tests

I will add these after i can test it all.

**Comment by Zaid-Ajaj** [comment]
Hi @Freymaurer, thanks  a lot for giving this a try! Testing new functionality should be really easy I believe because the test environment is already setup in this project. Simply add a couple of tests to the `tests` project like [this one](https://github.com/Zaid-Ajaj/Fable.SimpleXml/blob/master/test/Tests.fs#L527) and run the [live tests](https://github.com/Zaid-Ajaj/Fable.SimpleXml#running-the-tests-live) for a quick feedback loop 😉

**Comment by Freymaurer** [comment]
Ok will give it a try tomorrow. How can i reference the project dll in a .fsx? So i can play around with the functions first and also see what the results look like.

**Comment by Zaid-Ajaj** [comment]
You can't. These functions only work with Fable which doesn't support a REPL like F# on dotnet does

**Comment by Freymaurer** [comment]
Ok let me be honest, i tried to use the live tests but i have no idea what to do with this :sweat_smile::

![image](https://user-images.githubusercontent.com/39732517/103363338-2a108e80-4abb-11eb-800e-ebd7b2ac57d1.png)

Anyhow, i added tests and a short addition to the readme. Let me know what you think!

**Comment by Freymaurer** [comment]
So how is it going? Is something wrong with the PR? If so just tell me, i can change stuff 😅

**Comment by Zaid-Ajaj** [comment]
Sorry Kevin, I totally forgot about the PR 😓 I will have a good look at it later tonight

**Comment by Zaid-Ajaj** [comment]
Merged and published as part of v3.3.0 :rocket: please give it a try and let me know if you encounter any issues 🙏
